### PR TITLE
Release Wasmtime 0.34.0

### PR DESCRIPTION
**Pull request by wasmtime-publish**
This is an automated pull request from CI which is intended to
notify maintainers that it's time to release Wasmtime version
0.34.0. Version numbers have been bumped
in this PR automatically and the release process will automatically
enter the next stages once this PR is merged.

It's recommended that maintainers double-check that [RELEASES.md]
is up-to-date. If not please feel free to push to this PR any
modifications to the release notes. Additionally before merging it's
probably best to double-check the [release process] and make sure that
everything is ship-shape.

[RELEASES.md]: https://github.com/bytecodealliance/wasmtime/blob/main/RELEASES.md
[release process]: https://docs.wasmtime.dev/contributing-release-process.html

### CODE REVIEW

**Comment by alexcrichton** [comment]
@fitzgen IIRC there was an x86_64 issue you wanted to see fixed before this merged, is that still the case, and if so can you link it here?

**Comment by cfallin** [comment]
FWIW I think that we should probably get #3769 in (or some equivalent fix) before release -- otherwise the user can hit a panic if building with debuginfo and using cold blocks.

File: RELEASES.md [hunk 1]
```diff
@@ -2,6 +2,84 @@
 
 --------------------------------------------------------------------------------
 
+## 0.34.0
+
+Released 2022-02-08.
```
Comment (alexcrichton) [inline RELEASES.md:7]: This is a tentative date, but I'll update this if we get this merged today.

**Comment by alexcrichton** [comment]
I've added release notes here as well if someone wouldn't mind giving it a once-over as well

**Review by cfallin — APPROVED** [review]
r+ for the release notes, thanks for writing these up!

Someday all of us will remember to add to the release notes as we actually build the software :-)

**Comment by fitzgen** [comment]
> @fitzgen IIRC there was an x86_64 issue you wanted to see fixed before this merged, is that still the case, and if so can you link it here?

No issue on file yet. I need to check that the trap lowerings that have been ported to ISLE on x64 either don't need safepoints or implement safepoints on x64 for them if they do need them.

**Comment by fitzgen** [comment]
> > @fitzgen IIRC there was an x86_64 issue you wanted to see fixed before this merged, is that still the case, and if so can you link it here?
> 
> No issue on file yet. I need to check that the trap lowerings that have been ported to ISLE on x64 either don't need safepoints or implement safepoints on x64 for them if they do need them.

https://github.com/bytecodealliance/wasmtime/pull/3773

**Pull request merged**